Modules/Bridge/VtkGlue/test/itkVTKImageToImageFilterTest.cxx

```diff
@@ -33,15 +35,54 @@ int itkVTKImageToImageFilterTest(int, char*[])
   noise_source->SetMaximum(1.0);
   noise_source->Update();
 
+  auto input = noise_source->GetOutput();
+  input->SetSpacing(0.1, 2.0, 0.0);
+  input->SetOrigin(-0.1, -10, 0.0);
+  input->SetDirectionMatrix(0, 1, 0, -1, 0, 0, 0, 0, 1);
+  input->Print(std::cout);
+
   ConnectorType::Pointer connector = ConnectorType::New();
-  connector->SetInput(noise_source->GetOutput());
+  connector->SetInput(input);
   connector->Update();
 
-  connector->GetOutput()->Print(std::cout);
-  connector->GetImporter()->Print(std::cout);
-  connector->GetExporter()->Print(std::cout);
+  auto output = connector->GetOutput();
+  output->Print(std::cout);
+
+  for (int i = 0; i < dim; ++i)
+  {
+    if (output->GetLargestPossibleRegion().GetSize()[i] != input->GetDimensions()[i])
+    {
+      std::cerr << "Error: sizes do not match for component (" << i << ")." << std::endl;
+      return EXIT_FAILURE;
+    }
+    if (output->GetSpacing()[i] != input->GetSpacing()[i])
+    {
+      std::cerr << "Error: spacings do not match for component (" << i << ")." << std::endl;
+      return EXIT_FAILURE;
+    }
+    if (output->GetOrigin()[i] != input->GetOrigin()[i])
+    {
+      std::cerr << "Error: origins do not match for component (" << i << ")." << std::endl;
+      return EXIT_FAILURE;
+    }
+    for (int j = 0; j < dim; ++j)
+    {
+      if (output->GetDirection()[i][j] != input->GetDirectionMatrix()->GetData()[i*3+j])
+      {
+        std::cerr << "Error: directions do not match for component (" << i << "," << j << ")." << std::endl;
+        return EXIT_FAILURE;
+      }
+    }
+  }
 
-  connector->Print(std::cout);
+  // Try again with an invalid direction matrix and ensure it throws an exception
+  input->SetDirectionMatrix(0, 1, 0, 1, 0, 0, 1, 0, 0);
```

Comment:
Could you also test that it the matrix cannot be cropped, the appropriate exception is thrown?

Reply:
That's what this should be testing here.

Reply:
Looking at your matrix, it looked like your matrix is just not a great direction matrix as it is not a rotation matrix since the second and third rows are equal (determinant of your matrix is 0).

Reply:
Yeah I messed up while testing around, nice catch thanks.
I plan to have:
```
0 1 0
0 0 1
1 0 0
```